### PR TITLE
Version platform-specific files

### PR DESCRIPTION
**Pull request by wilzbach**
This allows to simplify the build system.

### CODE REVIEW

**Comment by wilzbach** [comment]
See also https://github.com/dlang/dlang.org/pull/2040 for the respective dlang.org cleanup.

**Comment by jacob-carlborg** [comment]
I think this is a bad idea. We should strive do the opposite: include all these platform specific files on all platforms and decide which object format/backend should be used at runtime (of the compiler). This will make it possible to cross-compile.

**Comment by WalterBright** [comment]
@jacob-carlborg I think you're right as an eventual goal, but in the meantime this is a small improvement.

File: src/dmd/libelf.d [hunk 1]
```diff
@@ -11,6 +11,9 @@
 module dmd.libelf;
 
 // Online documentation: https://dlang.org/phobos/dmd_libelf.html
+version(Windows) {}
+else version(OSX) {}
+else:
```
Comment (ibuclaw) [inline src/dmd/libelf.d]: Isn't there version identifiers for binary formats?

Comment (jacob-carlborg) [reply src/dmd/libelf.d]: As far as I can see there are `TARGET_*` for the target platform and `OBJ_*` for the object format. But the latter is set at runtime.

**Comment by jacob-carlborg** [comment]
@wilzbach you made it a lot more complicated for me with this PR. https://github.com/dlang/dmd/pull/7513.

**Pull request merged**